crypto/x509/x509_vfy.c

```diff
@@ -3456,7 +3456,7 @@ static int check_curve(X509 *cert)
         EVP_PKEY_get_int_param(pkey,
                                OSSL_PKEY_PARAM_EC_DECODED_FROM_EXPLICIT_PARAMS,
                                &val);
-    return ret < 0 ? ret : !val;
+    return ret == 0 ? -1 : !val;
```

Comment:
This is wrong. Do not change the condition, only the returned value.

Reply:
Thanks for your comment and sorry for the late response.

IMHO, if I understand correctly, the condition `ret < 0` does not work since `EVP_PKEY_get_int_param` returns 0 and 1 and no negative value.

```c
int EVP_PKEY_get_int_param(const EVP_PKEY *pkey, const char *key_name,
                           int *out)
{
    OSSL_PARAM params[2];

    if (key_name == NULL)
        return 0;

    params[0] = OSSL_PARAM_construct_int(key_name, out);
    params[1] = OSSL_PARAM_construct_end();
    return EVP_PKEY_get_params(pkey, params)
        && OSSL_PARAM_modified(params);
}
```

Or I misunderstand `only the returned value`? Does it mean change the return value of `EVP_PKEY_get_int_param` to make it return negative values? Thanks for your help again.

Reply:
TBH I am not familiar with this code and believe in your judgment. Do you mean to change it to `return ret == 0 ? 0 : !val;` or `return ret < 0 ? 0 : !val;`? I will change as your suggestions. Thank you. :-)